DataAccess/DataAccess.Postgre/PostgreAdapter.cpp

```diff
@@ -8,18 +8,18 @@ namespace DataAccess
 
         if (_instance == nullptr)
         {
-            _instance = std::shared_ptr<PostgreAdapter>(new PostgreAdapter(options.empty() ? _defaultOptions : options));
+            _instance = std::shared_ptr<PostgreAdapter>(new PostgreAdapter(options));
```

Comment:
Better change to make_shared

Reply:
That would be cool, but there is a problem enclosing the members of the PostgreAdapter class. When instantiating a smart pointer to a PostgreAdapter object, we can't access the private _connection member when the PostgreAdapter is constructed. Also, _connection is of type std::unique_ptr<pqxx::connection>. 

![image](https://user-images.githubusercontent.com/70116007/214091285-03f36dbb-a863-4a7c-bafa-e0c79e86beb4.png)